source/DataBase/DataBaseManager.cpp

```diff
@@ -1,34 +1,32 @@
 
 #include "DataBaseManager.hpp"
+
 #include "QueryOperationBuilder.hpp"
-#include "common/UserTable.hpp"
-namespace dataBase
-{
-    DataBaseManager::DataBaseManager(std::shared_ptr<interface::IDataBaseConnection> dataBaseConnection,
-                                     std::shared_ptr<interface::IQueryOperationBuilder> queryOperationBuilder)
-        : dataBaseConnection_(std::move(dataBaseConnection)), queryOperationBuilder_(std::move(queryOperationBuilder))
-    {
-    }
-
-    int DataBaseManager::executeOperation(const common::DataBaseRequest &requestData,
-                                          std::shared_ptr<common::ITable> records)
-    {
-        std::string result = queryOperationBuilder_->buildQuery(requestData);
-
-        if (requestData.operationType_ != common::DataBaseRequest::OperationType::Select)
-        {
-            dataBaseConnection_->executeQuery(result);
-            return static_cast<int>(common::DataBaseResponse::ResponseSuccess);
-        }
-
-        auto callback = [](void *data, int argc, char **argv, char **azColName) -> int
-        {
-            auto recordsPtr = *static_cast<std::shared_ptr<common::ITable> *>(data);
-            return recordsPtr->selectCallback(data, argc, argv, azColName);
-        };
-
-        dataBaseConnection_->executeQuery(result, callback, records);
-        return static_cast<int>(common::DataBaseResponse::ResponseSuccess);
-    }
+namespace dataBase {
+DataBaseManager::DataBaseManager(
+    std::shared_ptr<interface::IDataBaseConnection> dataBaseConnection,
+    std::shared_ptr<interface::IQueryOperationBuilder> queryOperationBuilder)
+    : dataBaseConnection_(std::move(dataBaseConnection)),
+      queryOperationBuilder_(std::move(queryOperationBuilder)) {}
+
+void DataBaseManager::executeOperation(
+    const common::DataBaseRequest &requestData) {
+  std::string query = queryOperationBuilder_->buildQuery(requestData);
+
+  if (requestData.operationType_ !=
+      common::DataBaseRequest::OperationType::Select) {
+    dataBaseConnection_->executeQuery(query);
+  }
+}
+
+std::vector<std::string> DataBaseManager::executeAndGetOperation(
+    const common::DataBaseRequest &requestData) {
+  std::string query = queryOperationBuilder_->buildQuery(requestData);
+
+  std::vector<std::string> result;
+
+  dataBaseConnection_->executeQueryAndGetData(query, result);
```

Comment:
Please change the API to return the `std::vector<std::string> result`, not pass it by `&`